### PR TITLE
NotOperatorWithSpaceFixer - NotOperatorWithSuccessorSpaceFixer - depr…

### PR DESCRIPTION
**Pull request by SpacePossum**
closes 

https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4674

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1851,6 +1852,12 @@ Choose from the list of available rules:
 
   Unary operators should be placed adjacent to their operands.
 
+  Configuration options:
+
+  - ``not_operator_space`` (``'leading_and_trailing'``, ``'no_trailing'``,
```
Comment (kubawerlos) [inline README.rst]: This got me confused as the new user of PHP CS Fixer - why only those 3 options? Where are `no_leading_and_trailing`, `no_leading` and `one_leading` options? Are they "worse" then added 3?

Wouldn't be more clear to have 2 boolean options `leading` and `trailing` with value `true|false|null` to cover all combination?

Comment (SpacePossum) [reply README.rst]: I moved the current logic of the fixers into one and made the difference in behavior available through these configuration options. I didn't spend time on investigating new or better options, so thanks for the suggestion!

Comment (kubawerlos) [reply README.rst]: Ok, and `no_trailing` is the default for new fixer, I missed that, now I get it 👍

File: src/Fixer/Operator/NotOperatorWithSuccessorSpaceFixer.php [hunk 2]
```diff
@@ -48,7 +50,15 @@ public function getDefinition()
      */
     public function getPriority()
```
Comment (julienfalque) [inline src/Fixer/Operator/NotOperatorWithSuccessorSpaceFixer.php]: This method can be dropped now.

File: src/Fixer/Operator/NotOperatorWithSpaceFixer.php [hunk 3]
```diff
@@ -48,7 +50,15 @@ public function getDefinition()
      */
     public function getPriority()
```
Comment (julienfalque) [inline src/Fixer/Operator/NotOperatorWithSpaceFixer.php]: This method can be dropped now.

File: src/Fixer/Operator/UnaryOperatorSpacesFixer.php [hunk 4]
```diff
@@ -75,4 +131,19 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
...
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver(
+            [
+                (new FixerOptionBuilder('not_operator_space', 'Space around logical `!` (`not`) operators.'))
```
Comment (julienfalque) [inline src/Fixer/Operator/UnaryOperatorSpacesFixer.php]: Wouldn't it be better to have an `operators` option supporting all possible operators like `binary_operator_spaces` has?

**Comment by julienfalque** [comment]
@SpacePossum deprecations cannot be added to a patch version, this must be merged into master.

**Comment by SpacePossum** [comment]
Im pushing this back to the discussion about the request so we can agree on what to do there, thanks all for the reviewing!

**Pull request closed without merging**